src/main/java/yilia/Parser.java

```diff
@@ -0,0 +1,89 @@
+package yilia;
+
+import yilia.command.*;
+import yilia.exception.*;
+
+/**
+ * Represents the parser to parse text to other meaning types.
+ */
+public class Parser {
+    /**
+     * parses the text into commands.
+     *
+     * @param text The text.
+     * @return The command.
+     * @throws YiliaException  If the command is unknown to Yilia.
+     * @throws NoIndexException  If the index is not given by users.
+     */
+    public static Command parse(String text) throws DescriptionEmptyException, KeywordMissingException,
```

Comment:
i feel like adding too many if-else statements here might make the code a little cluttered and affect readability, just something to consider!